ui/app/src/Utils/FormValidation.elm

```diff
@@ -28,19 +36,55 @@ fromResult result =
 type alias ValidatedField =
     { value : String
     , validationState : ValidationState
+    , rows : Int
+    }
+
+
+type alias Config =
+    { padding : Float
+    , lineHeight : Float
+    , minRows : Int
+    , maxRows : Int
+    }
+
+
+config : Config
+config =
+    { padding = 20
+    , lineHeight = 20
+    , minRows = 3
+    , maxRows = 15
     }
 
 
 initialField : String -> ValidatedField
 initialField value =
     { value = value
     , validationState = Initial
+    , rows = config.minRows
     }
 
 
 updateValue : String -> ValidatedField -> ValidatedField
 updateValue value field =
-    { field | value = value, validationState = Initial }
+    let
+        rows =
+            lines value
+                |> length
+                |> clamp config.minRows config.maxRows
+    in
+    { field | value = value, validationState = Initial, rows = rows }
+
+
+updateTextAreaHeight : Int -> ValidatedField -> ValidatedField
+updateTextAreaHeight scrollHeight field =
+    let
+        rows =
+            ((toFloat scrollHeight - config.padding) / config.lineHeight)
+                |> ceiling
+                |> clamp config.minRows config.maxRows
+    in
+    { field | rows = rows }
```

Comment:
I see that there are two ways to update `rows`, the first way is using the number of lines in the value and the second way is using the `scrollHeight`. If the first way is working, we don't really need the second way. We don't even need to extend `ValidatedField` to store `rows`, we only need to introduce a view for the comment field that calculates the rows based on the number of lines in the value.

Reply:
Thanks for your review.
That’s true.  Using scrollHeight is my old implementation of it. 
revised it

Reply:
@m-masataka sorry for not making this clear. Because `rows` can be calculated from the value, it doesn't need to be stored in model and can be done in the view function. The proposed change adds `rows` to every input, even though this is only used in the comment textarea.

What do you think of introducing a separate view function that would render the comment textarea and calculate `rows`?

Reply:
@w0rm Sorry, I misunderstood.
> Because rows can be calculated from the value, it doesn't need to be stored in the model and can be done in the view function.

Exactly.

> The proposed change adds rows to every input, even though this is only used in the comment textarea.
> What do you think of introducing a separate view function that would render the comment textarea and calculate the rows attribute?

I see. I think textarea must be separated. 
Do you mean that create new function similar to ``validatedField`` ? 
If so, I'll revise that soon.

Reply:
@m-masataka yeah, exactly! Just a separate function similar to `validatedField` for the comment textarea. The `rows` attribute only makes sense for `textarea` we don't need it for all regular input fields.